docs/explainer.md

```diff
@@ -9,6 +9,96 @@ See [Discourse](https://discourse.wicg.io/t/filling-the-remaining-gap-between-we
 User agents will need to carefully consider when to make the Raw Sockets API available to web applications,
 and what UI will be shown to the user.
 
+
+
+### Threat
+
+MITM attackers may inject sockets API calls into a web page.
+
+### Mitigation
+
+The API will only available in secure contexts (HTTPS).
+
+
+
+### Threat
+
+A web app might initiate connections without the user realizing.
+
+### Mitigation
+
+When initiating a connection, the user agent would show some UI.
+
+The user can be asked to specify a hostname or IP address, with an option to permit future connections to this host.
```

Comment:
Since this is the cornerstone of our mitigations, maybe go into some more detail here perhaps with a UX mock.

Reply:
Agree... this is the challenging bit. How do you explain what is going on here to a user and what the implications of getting pwned are? Or, alternatively, is this made scary enough that a user doesn't start filling out random things. 

The concern is that if you can send the right bits of information you can attack the hardware, rewrite firmware, and if you are motivated enough, destroy the hardware. 

I'm thinking of this small attack:
https://www.cybersecurity-insiders.com/cyber-attack-launched-on-150000-printers-working-worldwide/

But is a web page can trick a user to ssh into some server on a company network, that's a pretty scary prospect.

Reply:
Right, but what we should always keep in mind is that (as with all of these "scary" web APIs), we are not adding unheard of abilities to a completely safe vacuum; we are competing with native applications which have much less restricted network access.

If we can replace native apps that need network access with web apps that do so in a highly restrictive model (such as requiring explicit user entry of the destination host), that should severely limit the ability of those apps to conduct large-scale scanning.

Reply:
We don't want to give users the incorrect impression they would be safer installing a native app.

Mock added: [preview](https://ewilligers.github.io/raw-sockets/docs/explainer.html)

Reply:
Local apps may be safer:

- I can firewall local apps from accessing only required services. 

- Local apps cannot be updated on the fly as are web pages.

- I have local apps signed by key I trust.

- I can use strace to log what a local app does.

Is there a plan to provide these as well?